crates/astria-sequencer/src/app.rs

```diff
@@ -344,13 +351,15 @@ impl App {
     async fn execute_block_data(
         &mut self,
         txs: Vec<bytes::Bytes>,
+        mut remaining_cometbft_bytes: Option<usize>,
     ) -> (Vec<SignedTransaction>, Vec<bytes::Bytes>) {
         let mut signed_txs = Vec::with_capacity(txs.len());
         let mut validated_txs = Vec::with_capacity(txs.len());
         let mut block_sequence_data_bytes: usize = 0;
         let mut excluded_tx_count: usize = 0;
+        let txs_cached_length = txs.len();
 
-        for tx in txs {
+        for (i, tx) in txs.into_iter().enumerate() {
```

Comment:
The contents of this block look incredibly complex. Is there a way to move the loop body into a freestanding function and simplify? The explicit flow control using the `continue` and `break` statements could then be simplified using the `?` operator.